### PR TITLE
feat(RovingFocus): add support for up/down arrows and home/end buttons

### PR DESCRIPTION
**Pull request by Barsnes**

### CODE REVIEW

File: .changeset/bright-knives-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@digdir/designsystemet-react": patch
+---
+
+RovingIndex: add support for up/down arrows and home/end buttons
```
Comment (mimarz) [inline .changeset/bright-knives-remain.md:5]: This is what is going to be inlined in the changelog, so it needs to be conventional commit style.

File: packages/react/src/utilities/RovingFocus/RovingFocusItem.tsx [hunk 2]
```diff
@@ -52,12 +52,24 @@ export const RovingFocusItem = forwardRef<HTMLElement, RovingFocusItemProps>(
...
         let nextItem: RovingFocusElement | undefined;
 
-        if (e.key === 'ArrowRight') {
+        if (e.key === 'ArrowRight' || e.key === 'ArrowDown') {
           nextItem = getNextFocusableValue(items, focusValue);
+          e.preventDefault();
         }
 
-        if (e.key === 'ArrowLeft') {
+        if (e.key === 'ArrowLeft' || e.key === 'ArrowUp') {
```
Comment (mimarz) [inline packages/react/src/utilities/RovingFocus/RovingFocusItem.tsx:60]: This has an unintentional side-effect of focus moving left and right on up/down arrow keys on `Tabs` and `ToggleGroup` which are horisontal. Pretty sure this should not be the case.

Edit: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

**Review by mimarz — CHANGES_REQUESTED** [review]

File: packages/react/src/utilities/RovingFocus/RovingFocusItem.tsx (hunk 2 above)
Comment (Barsnes) [reply packages/react/src/utilities/RovingFocus/RovingFocusItem.tsx:60]: I thought this should always be the case, but we can make it a prop.

File: packages/react/src/utilities/RovingFocus/RovingFocusRoot.tsx [hunk 3]
```diff
@@ -21,6 +21,18 @@ type RovingFocusRootBaseProps = {
...
+   *
+   * @default false
+   */
+  verticalArrows?: boolean;
+  /**
+   * If `true`, the `RovingFocusItem` will be focusable with the up and down arrow keys.
+   *
+   * @default true
+   */
+  horizontalArrows?: boolean;
```
Comment (mimarz) [inline packages/react/src/utilities/RovingFocus/RovingFocusRoot.tsx:35]: Aren't these mutually exclusive? 🤔  

Suggest using a string literal for this instead; `orientation: 'vertical' | 'horizontal'`

**Review by mimarz — CHANGES_REQUESTED** [review]

File: packages/react/src/utilities/RovingFocus/RovingFocusRoot.tsx (hunk 3 above)
Comment (Barsnes) [reply packages/react/src/utilities/RovingFocus/RovingFocusRoot.tsx:35]: I don't believe they are, check out this; https://ui.shadcn.com/docs/components/toggle-group

Comment (mimarz) [reply packages/react/src/utilities/RovingFocus/RovingFocusRoot.tsx:35]: Either way we should still avoid multiple flags for the same thing as it leads to more complex bloated api and using multiple flags is often harder to maintain/read. 

To support both cases we can do something like `orientation: 'vertical' | 'horizontal' | 'ambiguous'` and then just change the switch case to check on the `orientation` prop. 

Then just check for left/right or up/down in the respective cases, like so; `if (['ArrowLeft', 'ArrowRight'].includes(e.key)) {…}`

**Review by mimarz — APPROVED** [review]

**Pull request merged**